Project_0.Altis/MessageReceiver/MessageReceiver.hpp

```diff
@@ -7,11 +7,16 @@
 
 #define MESSAGE_ID_INVALID -66.6
 
+MessageReceiver_getThread = {
+	params ["_object"];
+	GETV(CALLM0(_object, "getMessageLoop"), "scriptHandle")
+};
+
 // Macro to ensure it is being called in proper thread
 #ifdef OOP_ASSERT
 #define ASSERT_THREAD(objNameStr) \
 try { \
-	private _properThread = (GETV(CALLM0(objNameStr, "getMessageLoop"), "scriptHandle") isEqualTo _thisScript) || (isNil "_thisScript"); \
+	private _properThread = (isNil "_thisScript") or {(([objNameStr] call MessageReceiver_getThread) isEqualTo _thisScript)}; \
```

Comment:
Not really sure why we need another call here?

Reply:
not sure what you mean "another"?